HttpServer/HttpServer/HttpServer.cs

```diff
@@ -0,0 +1,596 @@
+﻿using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.IO;
+using System.IO.Compression;
+using System.Linq;
+using System.Net.Security;
+using System.Reflection;
+using System.Security.Cryptography.X509Certificates;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.ResponseCompression;
+using Microsoft.AspNetCore.Server.Kestrel.Https;
+using Microsoft.Extensions.DependencyInjection;
+using Neo.IO.Json;
+
+
+namespace Neo.Plugins.HttpServer
+{
+    using Interceptor = Action<IHttpOperationPayload>;
+
+    public class HttpServer : Plugin, IHttpServer
+    {
+        private const int MaxPostValue = 1024 * 1024 * 2;
+
+        // WEB HOST HANDLING THE SERVER
+        private IWebHost _host;
+
+        // SETTINGS PROVIDED BY JSON FILE
+        private HttpServerSettings _settings;
+
+        // BINDED OPERATIONS TO BE CALLED WHEN REQUESTS MATCH IT'S SIGNATURE
+        private IDictionary<string, IDictionary<string, OperationTargetAndMethod>> _operations = new Dictionary<string, IDictionary<string, OperationTargetAndMethod>>();
+
+        // INJECTED PARAMETERS FOR OPERATIONS: When a controller needs a parameter of a specific TYPE that was injected it will get it automatically
+        private IDictionary<Type, Func<HttpContext, object>> _specialParameterInjectors = new Dictionary<Type, Func<HttpContext, object>>();
+
+        // REQUEST INTERCEPTORS: Actions that will be called to handle the request, can read, modify and abort the request and response. Can be used to call the operation aswell
+        private List<Interceptor> _requestInterceptors = new List<Interceptor>();
+
+        // RESPONSE INTERCEPTORS: Actions that will be called to handle the response, can read and modify request and response
+        private List<Interceptor> _responseInterceptors = new List<Interceptor>();
+
+        public override string Name => "HttpServer";
+
+        public override void Configure()
+        {
+            _settings = new HttpServerSettings(GetConfiguration());
+            InjectSpecialParameter(ctx => ctx);
+            InjectSpecialParameter(ctx => _settings);
+
+            RegisterDefaultInterceptors();
+
+            Start();
+        }
+
+        public override void OnPluginsLoaded()
+        {
+            ConfigurePlugins();
+        }
+
+        public async void ConfigurePlugins()
+        {
+            await Task.Delay(500);
```

Comment:
I believe this isn't the best solution, but the app crashes when I dont wait for plugins initialization with this delay